Story starter animations

danielkao14 opened the pull request:
**pls run npm i in mobile and server to resolve all dependencies, right now the package lock.json is deprecated**
Removed old StoryStarterScreen files and created a new StoryStarterComponent. Since the Object, Plot Points, Setting, and Character Trait screen all have the same functionality, the StoryStarterComponent reduces some files by reusing code. Updated styling to match hi-fi after UT styling as of 5/16/2024.

Each screen has the title, route, and text color passed in as a prop in the `AppNavigation.jsx` file. The route prop is a string that is hardcoded to the corresponding GET api call for each screen: 
`      const randomItem = await axios.get(`${process.env.EXPO_PUBLIC_SERVER_URL}/${route}/get`, { timeout: 20000 });`

### Main screen:
https://cdn.discordapp.com/attachments/750854253920256060/1240880815114555453/IMG_5353.PNG?ex=66482bd0&is=6646da50&hm=4e087b8354f64ef89a6fc7b6814e4c8a778c0091acbf92668eb8cb0400e351ac&

### Plot point, Character traits, Setting, Object screens:
https://cdn.discordapp.com/attachments/750854253920256060/1240880815123202138/IMG_5354.PNG?ex=66482bd0&is=6646da50&hm=9d88e50331fd11d5cbeb5d314534da824efff438c527b16394844f7450d6d016&
https://cdn.discordapp.com/attachments/750854253920256060/1240880815240380497/IMG_5356.PNG?ex=66482bd0&is=6646da50&hm=e3127bf9724b7aa98919a2e5ca36838ec9ec041d7ad8145a72afeb234f82daaa&
https://cdn.discordapp.com/attachments/750854253920256060/1240880815106166845/IMG_5355.PNG?ex=66482bd0&is=6646da50&hm=952340041865f0e460460ad9aa14960a3b69eff5ea57cccff58a22f5fea57a46&
https://cdn.discordapp.com/attachments/750854253920256060/1240880814934327336/IMG_5357.PNG?ex=66482bd0&is=6646da50&hm=8db6bcb832c8139163ff13c850fc7e6168db4cc945b8833fa25aa466cd5f31b7&

danielkao14 committed:
story starters mid fi 3 rough draft

rough styling for mid fi 3 version of story starters

danielkao14 committed:
Remove header

Removes header for story starters, writing activities, mind and body, and how to screens

danielkao14 committed:
basic CSS for Story Starters

danielkao14 committed:
style button to linear gradient

danielkao14 committed:
update Story Starters main screen

danielkao14 committed:
Style storyStarter component

added handwriting quotes to the story starter component, updated styling for hi fi

danielkao14 committed:
add color to text, finish component

danielkao14 committed:
Add different background colors, adjust title text

Added different backgrounds depending on the type, adjusted text to match Hifi font

danielkao14 committed:
Remove old object, setting, trait, and plot point screens

removed deprecated screens, they're now streamlined into one component

danielkao14 committed:
Merge branch 'main' into story-starter-animations

danielkao14 committed:
remove comments

danielkao14 committed:
Remove unnecessary images

danielkao14 committed:
Fix header bug

removed line 127 that caused the headerShown: false to not work properly

cdebbaruah commented on mobile/Navigation/AppNavigation.jsx:
```diff
@@ -28,24 +25,40 @@ import EditPasswordScreen from '../Screens/EditPasswordScreen';
+      />
+      <StoryStarterStack.Screen
+        name="Objects"
+        children={() => (<StoryStarterComponent title="OBJECT" route="item" textColor="#7BAC8A" />)}
```
Try to avoid passing children as props!  Instead, it's better practice to nest the child components between the opening and closing tags of the parent component. Something like this should work:
```
      <StoryStarterStack.Screen
        name="Objects"
      >
        {() => (<StoryStarterComponent title="OBJECT" route="item" textColor="#7BAC8A" />)}
      </StoryStarterStack.Screen>
```

danielkao14 replied:
fixed!

cdebbaruah commented on mobile/Screens/StoryStarters/StoryStarterComponent.jsx:
Since this is a separate screen, I would recommend changing the file name to something other than `StoryStarterComponent` since we have a separate `Component` folder so things could get a little confusing!

danielkao14 replied:
fixed!

cdebbaruah reviewed (changes requested):
Amazing job, Daniel!! The story starters look great, and I love the reusability of the screens. Super minor eslint comment, but otherwise LGTM!

danielkao14 committed:
Merge branch 'main' into story-starter-animations

danielkao14 committed:
Fix eslint, rename file

Fixed ESlint warning by nesting children instead of passing as props. Renamed file StoryStarterComponent to StoryStarterPage to make the name distinct from the components folder.

cdebbaruah committed:
Merge branch 'main' into story-starter-animations
